src/features/context-menu/menu-providers.ts

```diff
@@ -0,0 +1,82 @@
+/********************************************************************************
+ * Copyright (c) 2019 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { injectable, multiInject, optional } from "inversify";
+
+import { MenuItem } from "./context-menu-service";
+import { SModelRoot } from "../../base/model/smodel";
+import { Point } from "../../utils/geometry";
+import { LabeledAction } from "../../base/actions/action";
+import { TYPES } from "../../base/types";
+import { isDeletable, DeleteElementAction } from "../edit/delete";
+import { isSelected } from "../select/model";
+
+export interface IContextMenuItemProvider {
+    getItems(root: Readonly<SModelRoot>, lastMousePosition?: Point): Promise<LabeledAction[]>;
+}
+
+@injectable()
+export class ContextMenuProviderRegistry implements IContextMenuItemProvider {
+
+    constructor(@multiInject(TYPES.IContextMenuProvider) @optional() protected menuProviders: IContextMenuItemProvider[] = []) { }
```

Comment:
Shouldn't that symbol be called `IContextMenuItemProvider` then?

Reply:
Oh, sure... good catch!